final sprint 7

neryklin opened the pull request:
1 step
Serializable

Commit:
final sprint 7
1 step
Serializable

Commit:
final sprint 7
2 step
depricated FileBackedSyncronizedTaskManagerTaskManager Serializable

add save and load CSV files

Umka7389 commented on src/FileBackedSyncronizedTaskManager.java:
```diff
@@ -0,0 +1,89 @@
+                ObjectInputStream objectInputStream = new ObjectInputStream(fileInputStream);) {
+            this.tasks = (HashMap) objectInputStream.readObject();
+        } catch (IOException e) {
+//            throw new ManagerSaveException(e.getMessage());
```
Закомментированный код не стоит оставлять и выбросить кастомный эксепшн выглядит хорошей идеей

neryklin replied:
Так минуточку :)
Вы не то проверили.
мое упущение что я оставил часть первичной реализации ТЗ через сериалазабл.
В данном случае оставил его только для того что бы малоли в дальнейшем надо будет его допилить но пометил как 
**@Deprecated**
public class FileBackedSyncronizedTaskManager extends InMemoryTaskManager implements TaskManager {

Umka7389 replied:
Ну я проверяю PR, какие вы изменения в нем сделали, те и проверяю)

Umka7389 commented on src/FileBackedSyncronizedTaskManager.java:
```diff
@@ -0,0 +1,89 @@
+        }
+    }
+
+    public class ManagerSaveException extends RuntimeException {
```
Эксепшены обычно выделяются в отдельный пакет, не говоря уже о классе

neryklin replied:
Так минуточку :)
Вы не то проверили.
мое упущение что я оставил часть первичной реализации ТЗ через сериалазабл.
В данном случае оставил его только для того что бы малоли в дальнейшем надо будет его допилить но пометил как
@deprecated
public class FileBackedSyncronizedTaskManager extends InMemoryTaskManager implements TaskManager {

neryklin replied:
Тем не менее в основном пакете была такая же проблема исправил.

Umka7389 commented on src/FileBackedSyncronizedTaskManager.java:
```diff
@@ -0,0 +1,89 @@
+        super.removeTaskFromId(id);
+        save();
+        return super.removeTaskFromId(id);
+        //  save();
```
Закомментированный код - проверьте по всему проекту

neryklin replied:
Вот тут согласен. Исправил.

Umka7389 reviewed (commented):
Молодец, хорошая работа! Замечания в комментариях

Commit:
final sprint 7
3 step
depricated FileBackedSyncronizedTaskManagerTaskManager Serializable

add save and load CSV files

fix comments

Umka7389 reviewed (commented):
Молодец, хорошая работа!
